### PR TITLE
WIP: integration of front with backend websockets

### PR DESCRIPTION
**Pull request by andrevigneaux**
there is still work to do in different parts, but more or less this sets the structure of how the communication and the passing of data should be handled

### CODE REVIEW

File: frontend/public/src/cards/card-mechanics.js [hunk 1]
```diff
@@ -15,10 +21,12 @@ export class CardMechanics {
   }
 
   matches(image) {
-    if (image.id === 1) {
-      return this.isGameActive();
+    if (gameState.middleCard.includes(image.id)) {
+      const scoreMessage = {};
```
Comment (andrevigneaux) [inline frontend/public/src/cards/card-mechanics.js:25]: todo

File: frontend/public/src/cards/card-mechanics.js [hunk 2]
```diff
@@ -15,10 +21,12 @@ export class CardMechanics {
   }
 
   matches(image) {
-    if (image.id === 1) {
-      return this.isGameActive();
+    if (gameState.middleCard.includes(image.id)) {
```
Comment (andrevigneaux) [inline frontend/public/src/cards/card-mechanics.js:24]: add && not blocked

File: frontend/scenes/GameScene.js [hunk 3]
```diff
@@ -18,58 +22,81 @@ export default class GameScene extends Phaser.Scene {
...
     });
     this.load.image("bg", "/assets/background.jpg");
-    this.fetchImages().catch(error => {
+    this.fetchImages().catch((error) => {
       console.error("Error fetching images:", error);
       this.errorMessage.show(error.message || "Error fetching images");
     });
   }
 
+  init(data) {
```
Comment (andrevigneaux) [inline frontend/scenes/GameScene.js:31]: more info will be passed here

File: frontend/scenes/GameScene.js [hunk 4]
```diff
@@ -18,58 +22,81 @@ export default class GameScene extends Phaser.Scene {
...
+  }
+
+  otherPlayerScore(newMiddleCard, player) {
+    this.cardMechanics.updateMiddleCardWithNewlyCreatedCard(
+      newMiddleCard,
+      player
+    );
   }
 
   gameEnd(winner) {
```
Comment (andrevigneaux) [inline frontend/scenes/GameScene.js:102]: todo: close ws connection from client side here (backend does it already but just in case)

**Review by jonty-kennedy-trendsales — APPROVED** [review]

**Pull request merged**